diff --git a/app/Http/Controllers/SellPosController.php b/app/Http/Controllers/SellPosController.php
index a9158e4..9d4fdfb 100755
--- a/app/Http/Controllers/SellPosController.php
+++ b/app/Http/Controllers/SellPosController.php
@@ -312,14 +312,22 @@ class SellPosController extends Controller
             $storeProduct2 = [];
             foreach($input['products'] as $productData)
             {
-                if($productData['category_id'] == 21)
+                if(isset($productData['category_id']))
                 {
-                    $storeProduct2[] = $productData;
+                    if($productData['category_id'] == 21)
+                    {
+                        $storeProduct2[] = $productData;
+                    }
+                    else
+                    {
+                        $storeProduct[] = $productData;
+                    }
                 }
                 else
                 {
                     $storeProduct[] = $productData;
                 }
+                
             }
             if(!empty($storeProduct))
             {
@@ -342,11 +350,15 @@ class SellPosController extends Controller
                 }
 
                 $input['is_quotation'] = 0;
+                
                 //status is send as quotation from Add sales screen.
-                if ($input['status'] == 'quotation') {
-                    $input['status'] = 'draft';
-                    $input['is_quotation'] = 1;
-                    
+                if(isset($input['status']))
+                {
+                    if ($input['status'] == 'quotation') {
+                        $input['status'] = 'draft';
+                        $input['is_quotation'] = 1;
+                        
+                    }
                 }
 
                 if (!empty($input['products'])) {
@@ -361,10 +373,17 @@ class SellPosController extends Controller
             
                     $user_id = $request->session()->get('user.id');
                     $commsn_agnt_setting = $request->session()->get('business.sales_cmsn_agnt');
-
+                    if(isset( $input['discount_type']))
+                    {
                     $discount = ['discount_type' => $input['discount_type'],
                                     'discount_amount' => $input['discount_amount']
                                 ];
+                    }
+                    else
+                    {
+                        $discount =0;
+                    }
+                    $input['tax_rate_id'] =0;
                     $invoice_total = $this->productUtil->calculateInvoiceTotal($input['products'], $input['tax_rate_id'], $discount);
 
                     DB::beginTransaction();
